core/src/main/kotlin/org/evomaster/core/search/service/SearchAlgorithm.kt

```diff
@@ -30,16 +30,45 @@ abstract class SearchAlgorithm<T> where T : Individual {
     @Inject
     protected lateinit var config: EMConfig
 
-
     @Inject(optional = true)
     private lateinit var mutator: Mutator<T>
 
+    private var lastSnapshot = 0
 
-    protected fun getMutatator() : Mutator<T> {
+    protected fun getMutator(): Mutator<T> {
         return mutator
     }
 
-    abstract fun search() : Solution<T>
+    abstract fun searchOnce()
```

Comment:
can u add JavaDoc to explain what is this one? I guess it is doing a single step in the search process